### PR TITLE
feat: support and fix slug match

### PR DESCRIPTION
**Pull request by yin1999**
reference: mdn/yari#6429

## test

image

### CODE REVIEW

File: src/cli.js [hunk 1]
```diff
@@ -145,14 +163,16 @@ program
       );
       progressBar.start(documents.count);
 
+      const slugPrefixLC = folderToSlugLC(folder);
```
Comment (yin1999) [inline src/cli.js:166]: such as `web/css/_colon_host`, the slug should be `web/css/:host` (in lowercase)

File: src/cli.js [hunk 2]
```diff
@@ -130,12 +144,16 @@ program
...
+      if (folder.length !== 0 && !folder.endsWith(path.sep)) { // if folder is specified, find folder only
+        folder += path.sep;
+      }
       console.info(
         `Starting HTML to Markdown conversion in ${options.mode} mode`
       );
       const documents = Document.findAll({
-        folderSearch: folder,
+        // replace '\' with '\\' to make this regexp works on Windows
+        folderSearch: os.platform() === "win32" ? folder.replace(/\\/g, "\\\\") : folder,
```
Comment (yin1999) [inline src/cli.js:156]:
```js
re = new RegExp('web\\\\api\\\\')

re.test('web\\api\\index.html') // true
```

image

File: src/cli.js [hunk 3]
```diff
@@ -130,12 +144,16 @@ program
   .argument("[folder]", "convert by folder")
   .action(
     tryOrExit(async ({ args, options }) => {
-      const folder = args.folder || "";
+      let folder = (args.folder || "").replace(/\\|\//g, path.sep); // correct path separator
+      if (folder.length !== 0 && !folder.endsWith(path.sep)) { // if folder is specified, find folder only
+        folder += path.sep;
```
Comment (yin1999) [inline src/cli.js:149]: `Document.findAll` will metch the filename, such as `web/api/index.html`, add a tailing slash to match folder

File: src/cli.js [hunk 4]
```diff
@@ -92,7 +94,7 @@ function saveProblemsReport(problems) {
     }
   }
   if (problemCount > 0) {
-    const reportFileName = `md-conversion-problems-report-${now.toISOString()}.md`;
+    const reportFileName = `md-conversion-problems-report-${now.toISOString().replace(/:/g, "_")}.md`;
```
Comment (yin1999) [inline src/cli.js:97]: windows file path is not support `:`

https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#naming-conventions

File: src/cli.js [hunk 5]
```diff
@@ -108,6 +110,18 @@ function buildLocaleMap(locale) {
   return localesMap;
 }
 
+// opposite of slugToFolder
+function folderToSlugLC(folder) {
```
Comment (queengooborg) [inline src/cli.js:114]:
```suggestion
function folderToSlug(folder) {
```

File: src/cli.js (hunk 1 above)
Comment (queengooborg) [inline src/cli.js:166]:
```suggestion
      const slugPrefix = folderToSlug(folder);
```

File: src/cli.js [hunk 6]
```diff
@@ -145,14 +163,16 @@ program
...
+
       const problems = new Map();
       try {
         for (let doc of documents.iter()) {
           progressBar.increment();
           if (
             doc.isMarkdown ||
             // findAll's folderSearch is fuzzy which we don't want here
-            !doc.metadata.slug.toLowerCase().startsWith(folder.toLowerCase())
+            !doc.metadata.slug.toLowerCase().startsWith(slugPrefixLC)
```
Comment (queengooborg) [inline src/cli.js:175]:
```suggestion
            !doc.metadata.slug.toLowerCase().startsWith(slugPrefix)
```

**Review by queengooborg — CHANGES_REQUESTED** [review]
Thanks!  Just a small nit: let's remove the "LC" suffix from the function and variable names for simplicity.

**Review by queengooborg — APPROVED** [review]

**Pull request merged**